### ISSUE TITLE
'with all children' incompatible with conditional components

### ISSUE THREAD

**Issue by dchambers** [bruderstein/unexpected-react-shallow#8]
If you have a JSX template that may or not have a child component, you might code it like this:

``` jsx
var child;
if(showChild) {
  child = <Child/>;
}
return (
  <Parent>
    {child}
  </Parent>
);
```

or, if you prefer the in-line aesthetic, like this:

``` jsx
return (
  <Parent>
  {showChild ? (
    <Child/>
  ) : null}
  </Parent>
);
```

In both cases, the rendered output will contain either an `undefined` or a `null` child component, which should just be ignored, as React itself does. At present however, if you use the `with all children` suffix on a template with `null` or `undefined` children, then the test will fail unnecessarily.

**Comment by bruderstein** [comment]
Thanks for reporting - I'll try and fix this ASAP.

**Comment by dchambers** [comment]
Lovely, thanks @bruderstein!

**Comment by bruderstein** [comment]
Just published v0.7.0 with this fixed

**Comment by dchambers** [comment]
Just tested it, works a treat, thanks @bruderstein :+1:
